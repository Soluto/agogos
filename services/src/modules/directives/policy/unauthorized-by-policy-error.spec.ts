import UnauthorizedByPolicyError from './unauthorized-by-policy-error';

describe('Unauthorized by policy', () => {
  test('For @policy directive', () => {
    const error = new UnauthorizedByPolicyError({ namespace: 'ns', name: 'policy' });
    expect(error).toMatchSnapshot();
  });

  test('For @policies directive', () => {
    const error = new UnauthorizedByPolicyError([
      new UnauthorizedByPolicyError({ namespace: 'ns-1', name: 'policy-1' }),
      new UnauthorizedByPolicyError({ namespace: 'ns-2', name: 'policy-2' }),
    ]);
    expect(error).toMatchSnapshot();
  });
});
