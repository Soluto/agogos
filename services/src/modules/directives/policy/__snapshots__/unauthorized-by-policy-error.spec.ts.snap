// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unauthorized by policy For @policies directive 1`] = `[UnauthorizedByPolicyError: Access to Foo.bar is unauthorized by policies: policy-1 (ns-1), policy-2 (ns-2)]`;

exports[`Unauthorized by policy For @policy directive 1`] = `[UnauthorizedByPolicyError: Access to Foo.bar is unauthorized by policy "policy" in namespace "ns"]`;
